### ISSUE TITLE
Add a function to pad regularly-spaced grids 

### ISSUE THREAD

**Issue by ivalaginja** [ehpor/hcipy#195]
Not possible for irregular grids, but this would be very useful for regularly spaced grids.

**Comment by syhaffert** [comment]
I do have that code somewhere. Let me dig it up and I will start a PR for it.

**Comment by ehpor** [comment]
A small addition to this: cropping. The `grid.subset()` method is kinda meant to do that, but it's currently not smart enough to detect that you've selected a region which is still separated and so can return a `Grid` with `SeparatedCoords`. Right now, it _always_ returns a `UnstructuredCoords` grid.

As a side question: what would be the arguments to this `Grid.pad()` function? I'd like to avoid anything to do with pixels, so it has to be physical distances in each of the axes.

**Comment by syhaffert** [comment]
Yes, I do use size units in my functions. So that should be okay. I might do this tomorrow because I have to figure out where I exactly used it....

**Comment by ehpor** [comment]
So exactly the same problem as me XD. I also have it somewhere, but I'll let you put in the PR.
